### DISCUSSION TITLE
Welcome to pyclustering Discussions!

### DISCUSSION THREAD

**Discussion by annoviko** [annoviko/pyclustering#666] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by aspitaleri** [comment]
Hi there,
I am comp chem/bioinfo and I am working on antimicrobial resistance/pathogen infections. I have used pyclustering to perform K-Means analysis but then I moved to UMAMP/HDBSCAN for my purpose. Anyway I still interested in pyclustering for K-Means/Agglomerative approaches. Thanks for your excellent job!

**Reply by annoviko** [reply]
Hello @aspitaleri,

Welcome! Nice to know that pyclustering is helping in your work. There are plans to implement HDBSCAN, but not enough hands for that. If you have any proposals, questions or even desire to contribute just let me know.

**Comment by Jayitha** [comment]
Hello,

I'm a first year Masters student at IIIT Hyderabad working in the Data Science and Analytics Center (DSAC). For my research work I found the pyclustering library very useful. I was looking for the implementation of the STING clustering algorithm, I see it's something you're working on for the next version. I was wondering if I could potentially attempt implementing the algorithm or anything else you need done for this library ? I want to gain more implementation experience.

**Comment by Raul9940** [comment]
Hello! I am a final year Physics BSc student currently doing my End of Degree Project about the Kuramoto Model. I am trying to learn how to use pyclustering because I think it would be really great to make the simulations I need with it. 
I am mainly focused in understanding sync.py and I have come across with the class sync_network, after looking to all the variables I am having trouble with the "frequency" one, it states it is the "multiplier of internal frequency of the oscillators" but I am unable to understand the meaning of this magnitude and how through its value I can set the frequency of each oscillator.

Thank you very much in advance.

**Comment by LiangJzzz** [comment]
Hello! Pyclustering was a remarkable work and I sincerely appreciated. When using it, I had some questions of Pyclustering. As we know, sklearn's kmeans used n_init to define initial centers and used max_iter to limit max iters. Did Pyclustering had similar parameters? In addition, did Pyclustering have internal conditions or convergence judgement? If the kmeans model is divergence, how to stop the iters.
Thank you very much and hope your reply soon!
